### PR TITLE
refactor: untangle ChainStore dependencies

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- replace `ChainStore` with `ChainIndex` in the VM,
- hoist `validate_tipsets` out of `StateManager`,
- delete `StateManager` and `ChainStore` from `forest-cli snapshot validate`,
- note: the Arcs can be cleaned up but that depends on other PRs that have yet to be merged.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/chain/store/index.rs [hunk 1]
```diff
@@ -145,6 +146,46 @@ impl<DB: Blockstore> ChainIndex<DB> {
...
+            }
+            if let Some(entry) = check_for_beacon_entry(&cur)? {
+                return Ok(entry);
+            }
+        }
+
+        if std::env::var(IGNORE_DRAND_VAR) == Ok("1".to_owned()) {
+            return Ok(BeaconEntry::new(
+                0,
+                vec![9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9],
```
Comment (LesnyRumcajs) [inline src/chain/store/index.rs:181]:
```suggestion
                vec![9; 16],
```

File: src/chain/store/index.rs [hunk 2]
```diff
@@ -145,6 +146,46 @@ impl<DB: Blockstore> ChainIndex<DB> {
             })
         })
     }
+
+    /// Finds the latest beacon entry given a tipset up to 20 tipsets behind
+    pub fn latest_beacon_entry(&self, ts: &Tipset) -> Result<BeaconEntry, Error> {
```
Comment (LesnyRumcajs) [inline src/chain/store/index.rs:151]: maybe we can use `anyhow` here, this `Error::Other` is not really more helpful. We might want to do it in a separate PR, though.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/chain/store/chain_store.rs [hunk 3]
```diff
@@ -358,7 +317,7 @@ where
         ts.blocks()
             .iter()
             .map(|b| {
-                let (usm, sm) = block_messages(self.blockstore(), b)?;
+                let (usm, sm) = block_messages(&db, b)?;
```
Comment (hanabi1224) [inline src/chain/store/chain_store.rs:320]:
```suggestion
                let (usm, sm) = block_messages(db, b)?;
```

**Review by hanabi1224 — APPROVED** [review]

File: src/chain/store/index.rs (hunk 2 above)
Comment (lemmih) [reply src/chain/store/index.rs:151]: Error handling will be cleaned up separately.

File: src/chain/store/chain_store.rs [hunk 4]
```diff
@@ -127,7 +126,7 @@ where
         chain_data_root: &Path,
     ) -> Result<Self> {
         let (publisher, _) = broadcast::channel(SINK_CAP);
-        let chain_index = ChainIndex::new(Arc::clone(&db));
+        let chain_index = Arc::new(ChainIndex::new(Arc::clone(&db)));
```
[the remark this answers is not in the document]
Comment (lemmih) [reply src/chain/store/chain_store.rs:129]: Using `Arc::clone` makes it explicit that you want to clone the `Arc` and not the underlying object. It makes the code easier to understand and harder to make mistakes.
